pkg/patch/patch.go

```diff
@@ -87,30 +150,107 @@ func removeIfNotDebug(workingFolder string) {
 	}
 }
 
-//nolint:lll
-func patchWithContext(ctx context.Context, ch chan error, image, reportFile, reportDirectory, platformSpecificErrors, patchedTag, suffix, workingFolder, scanner, format, output string, ignoreError, push bool, bkOpts buildkit.Opts) error {
+func patchWithContext(
+	ctx context.Context,
+	ch chan error,
+	image, reportFile, reportDirectory, platformSpecificErrors, patchedTag, suffix, workingFolder, scanner, format, output string,
+	ignoreError, push bool,
+	bkOpts buildkit.Opts,
+) error {
 	log.Debugf("Handling platform specific errors with %s", platformSpecificErrors)
+	if reportFile != "" && reportDirectory != "" {
+		return fmt.Errorf("both report file and directory provided, please provide only one")
+	}
 
+	// try report file
+	if reportFile != "" {
+		// check if reportFile exists
+		if _, err := os.Stat(reportFile); os.IsNotExist(err) {
+			return fmt.Errorf("report file %s does not exist", reportFile)
+		}
+		// check if reportFile is a file
+		f, err := os.Stat(reportFile)
+		if err != nil {
+			// handle common errors
+			if os.IsNotExist(err) {
+				return fmt.Errorf("report file %s does not exist", reportFile)
+			}
+			return fmt.Errorf("failed to stat report file %s: %w", reportFile, err)
+		}
+		if f.IsDir() {
+			return fmt.Errorf("report file %s is a directory, please provide a file", reportFile)
+		}
+		log.Debugf("Using report file: %s", reportFile)
+		platform := types.PatchPlatform{
+			Platform: platforms.Normalize(platforms.DefaultSpec()),
+		}
+		if platform.OS != "linux" {
+			platform.OS = "linux"
+		}
+		_, err = patchSingleArchImage(ctx, ch, image, reportFile, patchedTag, suffix, workingFolder, scanner, format, output, platform, ignoreError, push, bkOpts, false)
```

Comment:
should we save and output the PatchResult over here?

Reply:
Added some logging for the result here